src/main/java/build/buildfarm/common/distributed/DistributedMap.java

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buildfarm.common.distributed;
+
+import java.util.Map;
+
+/**
+ * @class DistributedMap
+ * @brief A distributed map.
+ * @details A distributed map is an implementation of a map data structure which can be shared
+ *     across a distributed system. For example, implementations of this interface might choose to
+ *     use redis, hazelcast, postgres, etc.
+ */
+public interface DistributedMap<DistributedClient> {
+
+  /**
+   * @brief Set key to hold the string value and set key to timeout after a given number of seconds.
+   * @details If the key already exists, then the value is replaced.
+   * @param client A client used by the implementation.
+   * @param key The name of the key.
+   * @param value The value for the key.
+   * @param timeout_s Timeout to expire the entry. (units: seconds (s))
+   * @note Overloaded.
+   */
+  void insert(DistributedClient client, String key, String value, int timeout_s);
+
+  /**
+   * @brief Set key to hold the string value and set key to timeout after a given number of seconds.
+   * @details If the key already exists, then the value is replaced.
+   * @param client A client used by the implementation.
+   * @param key The name of the key.
+   * @param value The value for the key.
+   * @param timeout_s Timeout to expire the entry. (units: seconds (s))
+   * @note Overloaded.
+   */
+  void insert(DistributedClient client, String key, String value, long timeout_s);
```

Comment:
why is this method necessary? you anticipate exceeding a billenium for an insert timeout?

Reply:
It looks like clients were often calling with longs and this was done to avoid updating all the callers.  I agree we should remove this and tighten the interface.

Reply:
this API has been removed.